src/main/java/org/apache/maven/plugins/clean/Cleaner.java

```diff
@@ -115,9 +135,102 @@ public void delete( File basedir, Selector selector, boolean followSymlinks, boo
 
         File file = followSymlinks ? basedir : basedir.getCanonicalFile();
 
+        if ( selector == null && !followSymlinks && fastDir != null )
+        {
+            // If anything wrong happens, we'll just use the usual deletion mechanism
+            if ( fastDelete( file ) )
+            {
+                return;
+            }
+        }
+
         delete( file, "", selector, followSymlinks, failOnError, retryOnError );
     }
 
+    private boolean fastDelete( File baseDir )
+    {
+        // Handle the case where we use ${maven.multiModuleProjectDirectory}/target/.clean for example
+        if ( fastDir.getAbsolutePath().startsWith( baseDir.getAbsolutePath() ) )
+        {
+            try
+            {
+                File tmpDir = createTempDir( baseDir.getParentFile(), baseDir.getName() + "." );
+                try
+                {
+                    Files.move( baseDir.toPath(), tmpDir.toPath(), StandardCopyOption.ATOMIC_MOVE );
+                    if ( session != null )
+                    {
+                        session.getRequest().getData().put( LAST_DIRECTORY_TO_DELETE, baseDir );
+                    }
+                    baseDir = tmpDir;
+                }
+                catch ( IOException e )
+                {
+                    tmpDir.delete();
+                    if ( logDebug != null )
+                    {
+                        logDebug.log( "Unable to fast delete directory: " + e );
+                    }
+                    return false;
+                }
+            }
+            catch ( IOException e )
+            {
+                if ( logDebug != null )
+                {
+                    logDebug.log( "Unable to fast delete directory: " + e );
+                }
+                return false;
+            }
+        }
+        // Create fastDir and the needed parents if needed
+        if ( fastDir.mkdirs() || fastDir.isDirectory() )
+        {
+            try
+            {
+                File tmpDir = createTempDir( fastDir, "" );
+                File dstDir = new File( tmpDir, baseDir.getName() );
+                // Note that by specifying the ATOMIC_MOVE, we expect an exception to be thrown
+                // if the path leads to a directory on another mountpoint.  If this is the case
+                // or any other exception occurs, an exception will be thrown in which case
+                // the method will return false and the usual deletion will be performed.
+                Files.move( baseDir.toPath(), dstDir.toPath(), StandardCopyOption.ATOMIC_MOVE );
+                BackgroundCleaner.delete( this, tmpDir );
+                return true;
+            }
+            catch ( IOException e )
+            {
+                if ( logDebug != null )
+                {
+                    logDebug.log( "Unable to fast delete directory: " + e );
```

Comment:
"Unable to fast delete directory: " + e

Reply:
I tried, but the `Cleaner` class defines its own logger interface to wrap the one passed in the constructor:
https://github.com/apache/maven-clean-plugin/blob/68c3aa67d04b02a6ee8d657146c0506c66a795d0/src/main/java/org/apache/maven/plugins/clean/Cleaner.java#L395-L400

Reply:
Shit, then please add a comment that this is intentional